src/types.rs

```diff
@@ -212,6 +242,25 @@ impl<'a> Token<'a> {
         self.raw.name_id
     }
 
+    /// Given some minified source this returns the most likely minified name.
+    ///
+    /// Note that this scans for identifiers in the source file so in some cases it can happen that
+    /// values are returned that are not actually names.  For instance a token that points to a
+    /// keyword will return the keyword.  This is done because it is not always possible to tell
+    /// keywords from non keywords without parsing the entire source.
+    pub fn get_minified_name<'b>(&self, source: &'b str) -> Option<&'b str> {
```

Comment:
the idea here is that we get the original token at the token location from the minified file. After talking to some browser people it turns out the columns are neither unicode characters nor bytes but UCS2 offsets which is why I use this wtf16_slice function to offset the string. I then parse one javascript identifier to the right from that location and ignore spaces because at least uglify includes spaces in the token -.-